### PR TITLE
lockdown: Propagate lower level errors to callers instead of just returning unknown error

### PR DESCRIPTION
**Pull request by zbalaton**
Small patch to allow callers to get a better error code instead of just a generic error to know what went wrong.

### CODE REVIEW

**Comment by zbalaton** [comment]
I've updated this to current HEAD now and also included some other small fixes. Will this be merged now?

File: src/lockdown.c [hunk 1]
```diff
@@ -722,6 +733,9 @@ LIBIMOBILEDEVICE_API lockdownd_error_t lockdownd_client_new_with_handshake(idevi
 		ret = lockdownd_pair(client_loc, NULL);
 		if (LOCKDOWN_E_SUCCESS == ret) {
 			ret = lockdownd_validate_pair(client_loc, NULL);
+			if (ret == LOCKDOWN_E_MUX_ERROR) {
+				ret = LOCKDOWN_E_INVALID_HOST_ID;
+			}
```
Comment (nikias) [inline src/lockdown.c]: why does mux error mean invalid host id?

**Comment by nikias** [comment]
Looks good to me, except I was wondering about the if (ret == LOCKDOWN_E_MUX_ERROR) ret = LOCKDOWN_E_INVALID_HOST_ID is the right way to do it.

**Comment by zbalaton** [comment]
LOCKDOWN_E_INVALID_HOST_ID is normally returned when pairing should be tried again. When the trust button is tapped, the device breaks connection which will result in MUX_ERROR but this at this point means that pairing was successful and should be retried so the error is translated to the error code which is normally returned for this case in other places. Using other error code would need clients already checking for INVALID_HOST_ID to be changed to also check for that other error code so this is the right way for compatibility.

**Comment by zbalaton** [comment]
I've removed the disputed hunk converting LOCKDOWN_E_MUX_ERROR to LOCKDOWN_E_INVALID_HOST_ID on pairing failure so the rest could be merged. We can consider this change separately in subsequent commit. Anything else holding this back?

**Comment by nikias** [comment]
No, I will merge it.

**Pull request merged**